twistr/game.go

```diff
@@ -58,12 +58,26 @@ func SelectShuffle(d *Deck) []Card {
 	return d.Shuffle()
 }
 
-func SelectCard(s *State, player Aff) (c Card) {
+func SelectCard(s *State, player Aff, blacklist ...CardId) (c Card) {
 	canPlayChina := s.ChinaCardPlayer == player && s.ChinaCardFaceUp
 	choices := make([]string, len(s.Hands[player].Cards))
+
+	isBlacklisted := func(c Card) bool {
+		for _, bl := range blacklist {
+			if c.Id == bl {
+				return true
+			}
+		}
+		return false
+	}
+
 	for i, c := range s.Hands[player].Cards {
```

Comment:
the original for-loop relies copies the card slice into the card name slice by index.
if you're skipping blacklisted cards, the card name slice will have gaps now. just use append?